pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -681,7 +687,11 @@ func (cloud *CloudProvider) waitForZonalOp(ctx context.Context, opName string, z
 	// The v1 API can query for v1, alpha, or beta operations.
 	svc := cloud.service
 	project := cloud.project
-	return wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+	timeout := OpTimeout
+	if deadline, ok := ctx.Deadline(); ok {
+		timeout = time.Until(deadline)
+	}
+	return wait.Poll(OpInterval, timeout, func() (bool, error) {
 		pollOp, err := svc.ZoneOperations.Get(project, zone, opName).Context(ctx).Do()
```

Comment:
I think it would make sense to separate this call to another wait block. So that we can make sure the operation has been created successfully. And that can use a smaller timeout to prevent a long timeout.